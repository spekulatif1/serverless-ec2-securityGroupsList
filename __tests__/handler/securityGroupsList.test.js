'use strict'

const {
	index: securityGroupsHandler
} = require('../../handler/securityGroupsList');
const {
	EC2Client,
	DescribeSecurityGroupsCommand
} = require("@aws-sdk/client-ec2");
const { mockClient } = require("aws-sdk-client-mock");
const ec2Mock = mockClient(EC2Client);
const {
	awsSecurityGroupsListResponse
} = require('../../data/sample');

describe('securityGroupsHandler', () => {
	beforeEach(() => {
		ec2Mock.reset();
	});

	test('securityGroupsHandler.HTTP.200', async() => {
		ec2Mock
			.on(DescribeSecurityGroupsCommand, { NextToken: undefined })
			.resolves(awsSecurityGroupsListResponse)

		const successResponse = await securityGroupsHandler();
		expect(successResponse).toMatchSnapshot();
	});

	test('securityGroupsHandler.HTTP.500', async() => {
		ec2Mock
			.on(DescribeSecurityGroupsCommand, { NextToken: undefined })
			.rejects({
				Code: 'AuthFailure',
				'$fault': 'client',
				'$metadata': {
					httpStatusCode: 401,
					requestId: 'b02554e9-82a9-4b63-8c75-6eae22f7446e',
					extendedRequestId: undefined,
					cfId: undefined,
					attempts: 3,
					totalRetryDelay: 564
				}
			})

		const successResponse = await securityGroupsHandler();
		expect(successResponse).toMatchSnapshot();
	});
});
