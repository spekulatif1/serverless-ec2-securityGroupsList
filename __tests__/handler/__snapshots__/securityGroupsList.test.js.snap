// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`securityGroupsHandler securityGroupsHandler.HTTP.200 1`] = `
Object {
  "body": "{\\"data\\":[{\\"Description\\":\\"TrendMicro-Assignment-Test\\",\\"GroupName\\":\\"TrendMicro-Assignment-Test\\",\\"IpPermissions\\":[{\\"FromPort\\":80,\\"IpProtocol\\":\\"tcp\\",\\"IpRanges\\":[{\\"CidrIp\\":\\"0.0.0.0/0\\"}],\\"Ipv6Ranges\\":[{\\"CidrIpv6\\":\\"::/0\\"}],\\"PrefixListIds\\":[],\\"ToPort\\":80,\\"UserIdGroupPairs\\":[]}],\\"OwnerId\\":\\"TrendMicro\\",\\"GroupId\\":\\"sg-trendMicro\\",\\"IpPermissionsEgress\\":[{\\"IpProtocol\\":\\"-1\\",\\"IpRanges\\":[{\\"CidrIp\\":\\"0.0.0.0/0\\"}],\\"Ipv6Ranges\\":[],\\"PrefixListIds\\":[],\\"UserIdGroupPairs\\":[]}],\\"VpcId\\":\\"vpc-trendMicro\\"}]}",
  "statusCode": 200,
}
`;

exports[`securityGroupsHandler securityGroupsHandler.HTTP.500 1`] = `
Object {
  "body": "{\\"errors\\":[{\\"detail\\":\\"\\",\\"title\\":\\"Internal Server Error\\"}]}",
  "statusCode": 500,
}
`;
